source/linux/Foundational_Components/Kernel/Kernel_Drivers/UBIFS.rst

```diff
@@ -0,0 +1,403 @@
+
+.. note::
+
+   Download and extract ``tisdk-default-image-*-evm.rootfs.tar.xz`` for
+   QSPI/OSPI/GPMC NAND.
+
+   Download and extract ``tisdk-tiny-image-*-evm.rootfs.tar.xz`` for QSPI/OSPI
+   NOR.
+
+The ``rootfs`` partition information
+
+   OSPI NOR
+
+   .. code-block:: console
+
+      # EVM Linux
+
+      $ mtdinfo /dev/mtd5
+      mtd5
+      Name:                           ospi.rootfs
+      Type:                           nor
+      Eraseblock size:                262144 bytes, 256.0 KiB
+      Amount of eraseblocks:          223 (58458112 bytes, 55.7 MiB)
+      Minimum input/output unit size: 16 bytes
+      Sub-page size:                  16 bytes
+      Character device major/minor:   90:10
+      Bad blocks are allowed:         false
+      Device is writable:             true
+
+   OSPI NAND
+
+   .. code-block:: console
+
+      # EVM Linux
+
+      $ mtdinfo /dev/mtd5
+      mtd5
+      Name:                           ospi_nand.rootfs
+      Type:                           nand
+      Eraseblock size:                262144 bytes, 256.0 KiB
+      Amount of eraseblocks:          383 (100401152 bytes, 95.7 MiB)
+      Minimum input/output unit size: 4096 bytes
+      Sub-page size:                  4096 bytes
+      OOB size:                       128 bytes
+      Character device major/minor:   90:10
+      Bad blocks are allowed:         true
+      Device is writable:             true
+
+   GPMC NAND
+
+   .. code-block:: console
+
+      # EVM Linux
+
+      $ mtdinfo /dev/mtd7
+      mtd7
+      Name:                           NAND.file-system
+      Type:                           nand
+      Eraseblock size:                262144 bytes, 256.0 KiB
+      Amount of eraseblocks:          4054 (1062731776 bytes, 1013.5 MiB)
+      Minimum input/output unit size: 4096 bytes
+      Sub-page size:                  1024 bytes
+      OOB size:                       256 bytes
+      Character device major/minor:   90:14
+      Bad blocks are allowed:         true
+      Device is writable:             true
+
+
+.. rubric:: Step 2
+   :name: linux-building-ubi-file-system-step2
+
+Create a file named :file:`ubinize.cfg` and add the contents below. Note that
+this configuration file can be outside the root directory.
+
+.. code-block:: text
+
+   [ubifs]
+   mode=ubi
+   image=rootfs.ubifs
+   vol_id=0
+   vol_type=dynamic
+   vol_name=rootfs
+   vol_flags=autoresize
+
+.. rubric:: Step 3
+   :name: linux-building-ubi-file-system-step3
+
+:command:`mkfs.ubifs` is used to create UBI File System image, which is
+specifically generated for a flash memory device, like Serial NOR, NAND and
+Parallel NAND.
+
+Syntax
+
+.. code-block:: console
+
+   mkfs.ubifs -r <root_dir> -o <output_image> [options]
+
+Some key options to use:
+
+1. ``-m <min_io_size>``: specifies the minimum I/O size (in bytes) of the flash
+   device.
+
+.. code-block:: text
+
+   The 'Minimum input/output unit size' value in mtdinfo.
+
```

Comment:
To add context to that, content associated with a list entry should be indented to fall under that list entry. If it's not indented it instead indicates the end of the list and that the next entry is actually the start of a new list.

Reply:
Oh, yes. Thank you, Randolph!!

Updated.